src/Configuration.php

```diff
@@ -353,13 +360,29 @@ public function setAppType($type)
     }
 
     /**
+     * Set the fallback application type.
+     *
+     * This is should be used only by libraries to set an fallback app type.
+     *
+     * @param string|null $type the fallback type
+     *
+     * @return $this
+     */
+    public function setFallbackType($type)
+    {
+        $this->fallbackType = $type;
+
+        return $this;
+    }
+
+    /**
      * Get the application data.
      *
      * @return array
      */
     public function getAppData()
     {
-        return array_merge(['type' => php_sapi_name(), 'releaseStage' => 'production'], array_filter($this->appData));
+        return array_merge(['type' => $this->fallbackType ?: php_sapi_name(), 'releaseStage' => 'production'], array_filter($this->appData));
```

Comment:
Yeh, so the array_filter means that if the user hasn't set an app type, then the stuff from the LHS of the array merge will replace it, and the ternary part just means we use the sapi name if there's no library set fallback.